Code/Tools/AssetProcessor/native/utilities/assetUtils.cpp

```diff
@@ -870,23 +870,40 @@ namespace AssetUtilities
         return true;
     }
 
-    QString StripAssetPlatform(AZStd::string_view relativeProductPath)
+    AZStd::string_view StripAssetPlatformNoCopy(AZStd::string_view relativeProductPath)
     {
         // Skip over the assetPlatform path segment if it is matches one of the platform defaults
         // Otherwise return the path unchanged
-        AZStd::string_view strippedProductPath{ relativeProductPath };
-        if (AZStd::optional pathSegment = AZ::StringFunc::TokenizeNext(strippedProductPath, AZ_CORRECT_AND_WRONG_FILESYSTEM_SEPARATOR);
-            pathSegment.has_value())
+
+        if (size_t pos = relativeProductPath.find_first_of(AZ_CORRECT_AND_WRONG_FILESYSTEM_SEPARATOR); pos == AZStd::string_view::npos)
         {
-            AZ::IO::FixedMaxPathString assetPlatformSegmentLower{ *pathSegment };
-            AZStd::to_lower(assetPlatformSegmentLower.begin(), assetPlatformSegmentLower.end());
-            if (AzFramework::PlatformHelper::GetPlatformIdFromName(assetPlatformSegmentLower) != AzFramework::PlatformId::Invalid)
+            // There's no separators, just return the path as is
+            return relativeProductPath;
+        }
+        else
+        {
+            // Grab the contents to the left of the separator and see if it matches a known platform string
+            // If it does, we'll return the right side, otherwise we assume there's no platform and return the whole string
+            AZStd::string_view leftSide{ relativeProductPath.data(), pos };
+
+            if (AzFramework::PlatformHelper::GetPlatformIdFromName(leftSide) != AzFramework::PlatformId::Invalid)
             {
-                return QString::fromUtf8(strippedProductPath.data(), aznumeric_cast<int>(strippedProductPath.size()));
+                AZStd::string_view rightSide{ relativeProductPath };
+                // Strip off all previous characters before the delimiter plus the delimiter itself
+                rightSide.remove_prefix(pos + 1);
+
+                return rightSide;
             }
+
+            return relativeProductPath;
         }
```

Comment:
FYI:: You can use `AZ::StringFunc::TokenizeNext` to split the path on the first separator

```
AZStd::string_view originalPath = relativeProductPath;
AZStd::optional firstPathSegment = AZ::StringFunc::TokenizeNext(relativeProductPath, AZ_CORRECT_AND_WRONG_FILESYSTEM_SEPARATOR);

if (AzFramework::PlatformHelper::GetPlatformIdFromName(*firstPathSegment) != AzFramework::PlatformId::Invalid)
{
    return relativeProductPath;
}

return originalPath;
```

Reply:
Updated